HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1619,15 +1788,54 @@ def self.apply_infiltration_and_ventilation_fans(model, weather, vent_fans_mech,
+    infil_program.addLine("Set #{remaining_oa_name} = Qfan - QWHV") # The remaining oa air loads need to be addressed
+    infil_program = apply_fan_load_to_infil_program(model: model, infil_program: infil_program, oa_flow_rate_var_name: remaining_oa_name, tot_flow_rate_var_name: remaining_oa_name, vent_mech_sens_eff: 0.0, vent_mech_lat_eff: 0.0, fan_sens_load_actuator: fan_sens_load_actuator, fan_lat_load_actuator: fan_lat_load_actuator, preconditioned: false)
 
     program_calling_manager = OpenStudio::Model::EnergyManagementSystemProgramCallingManager.new(model)
     program_calling_manager.setName("#{infil_program.name} calling manager")
     program_calling_manager.setCallingPoint('BeginTimestepBeforePredictor')
     program_calling_manager.addProgram(infil_program)
   end
 
+  def self.apply_infiltration_and_ventilation_fans(model, weather, vent_fans_mech, vent_fans_kitchen, vent_fans_bath, has_flue_chimney, air_infils, vented_attic, vented_crawl)
+    # Get living space infiltration
+    living_ach50 = nil
+    living_const_ach = nil
+    air_infils.each do |air_infil|
+      if (air_infil.unit_of_measure == HPXML::UnitsACH) && !air_infil.house_pressure.nil?
+        living_ach = air_infil.air_leakage
+        living_ach50 = calc_air_leakage_at_diff_pressure(0.65, living_ach, air_infil.house_pressure, 50.0)
+      elsif (air_infil.unit_of_measure == HPXML::UnitsCFM) && !air_infil.house_pressure.nil?
+        living_ach = air_infil.air_leakage * 60.0 / @infil_volume # Convert CFM to ACH
+        living_ach50 = calc_air_leakage_at_diff_pressure(0.65, living_ach, air_infil.house_pressure, 50.0)
+      elsif air_infil.unit_of_measure == HPXML::UnitsACHNatural
+        if @apply_ashrae140_assumptions
+          living_const_ach = air_infil.air_leakage
+        else
+          sla = get_infiltration_SLA_from_ACH(air_infil.air_leakage, @infil_height, weather)
+          living_ach50 = get_infiltration_ACH50_from_SLA(sla, 0.65, @cfa, @infil_volume)
+        end
+      end
+    end
+
+    # Infiltration for unconditioned spaces
+    apply_infiltration_to_garage(model, weather, living_ach50)
+    apply_infiltration_to_unconditioned_basement(model, weather)
+    apply_infiltration_to_vented_crawlspace(model, weather, vented_crawl)
+    apply_infiltration_to_unvented_crawlspace(model, weather)
+    apply_infiltration_to_vented_attic(model, weather, vented_attic)
+    apply_infiltration_to_unvented_attic(model, weather)
+
+    # Local ventilation
+    range_sch_sensors_map = apply_local_ventilation(model, vent_fans_kitchen, Constants.ObjectNameMechanicalVentilationRangeFan)
+    bath_sch_sensors_map = apply_local_ventilation(model, vent_fans_bath, Constants.ObjectNameMechanicalVentilationBathFan)
+
+    # Get mechanical ventilation
+    apply_mechanical_ventilation(model, vent_fans_mech, living_ach50, living_const_ach, weather, vent_fans_kitchen, vent_fans_bath, range_sch_sensors_map, bath_sch_sensors_map, has_flue_chimney)
```

Comment:
I combined the mech vent part into another method. It started with a feeling that it will make things more clear by separating mech vent out, but ended up with too many variables (highly intertwined with others) in this method, not sure whether it helps a lot on clearing things up.